### ISSUE TITLE
[PATCH] OrbitalCameraSystem.detach() — undocumented early-return on null canvas

### ISSUE THREAD

**Issue by ormidales** [ormidales/microgl#296]
**Category**: documentation
**Severity**: patch
**File(s)**: `src/core/ecs/systems/OrbitalCameraSystem.ts` (detach method)

#### Description
`detach()` silently returns when `this.canvas` is `null` (i.e. when `attach()` was never called or `detach()` is called a second time). This is the correct behaviour, but it is not mentioned in the method's JSDoc. Callers who miss this may add unnecessary null-guards or be surprised by the no-op.

#### Problematic code example
```ts
detach(): void {
  if (!this.canvas) return;
  // ... remove event listeners
  this.canvas = null;
}
```

#### Suggested fix
Add a brief JSDoc comment documenting the idempotent behaviour:

```ts
/**
 * Remove all event listeners registered by {@link attach} and release the
 * canvas reference. Safe to call even when `attach` was never invoked or
 * after a previous `detach` call (no-op in both cases).
 */
detach(): void {
```

#### Acceptance criteria
- [ ] JSDoc for `detach()` states the method is idempotent / safe to call multiple times
- [ ] Corresponding `attach()` JSDoc cross-references `detach()` via `{@link detach}`
